src/tvdb/client.py

```diff
@@ -0,0 +1,192 @@
+from abc import ABC, abstractmethod
+from typing import Any, ClassVar, Literal, overload, override
+
+import aiohttp
+from yarl import URL
+
+from src.settings import TVDB_API_KEY
+from src.tvdb.generated_models import (
+    MovieBaseRecord,
+    MovieExtendedRecord,
+    SearchResult,
+    SeriesBaseRecord,
+    SeriesExtendedRecord,
+)
+from src.tvdb.models import MovieResponse, SearchResponse, SeriesResponse
+from src.utils.log import get_logger
+
+log = get_logger(__name__)
+
+type JSON_DATA = dict[str, JSON_DATA] | list[JSON_DATA] | str | int | float | bool | None  # noice
+
+
+class _Media(ABC):
+    def __init__(self, client: "TvdbClient"):
+        self.client = client
+        self._id: int | None = None
+
+    @property
+    def id(self) -> int | str | None:
```

Comment:
In the setter, you're always converting this to an int, why does the return type indicate that this can also be a `str`?

Reply:
Because pyright was yelling at me and they had to be the same.

Reply:
Oh I see, yeah, in general, you don't want to allow setting a property to a different value than what it'll return, since it could be confusing to the user, who doesn't know about the inner logic inside of that property setter. In this case, either only allow `int`, or just ignore the `reportPropertyTypeMismatch`.

Then again, if you'll rework these classes anyways, this will likely just become a parameter for a function, where this won't be an issue.